### ISSUE TITLE
Agent reached an undesired plateau 

### ISSUE THREAD

**Issue by firepro20** [Unity-Technologies/ml-agents#2807]
I am training an agent to avoid spikes and collect health items through a maze full of spikes. 

The first 1hr 30 mins it made some progress and collected 3/4 of the total collectibles. However, it has been stuck for 30 mins in the same corner and a mean reward of 0.

I am providing as input vectors, 2 spike positions which are populated on triggerenter, the nearest collectible item when this enters the trigger, the agent position, rBody.x and rBody.z which will control. In total 14 vector observations.

The scene is bound by walls, several spikes on a plane and also collectible items. I am giving the following rewards on collision enter

```
if (collision.gameObject.CompareTag("Hazard"))
        {
            SetReward(-0.5f); // divide 1/noOfSpikes
            Done();
            GameController.Instance.ReceiveDamage(gameObject, healthOnHazard);
        }
        if (collision.gameObject.CompareTag("Collectible"))
        {
            Debug.LogError("Consumed Health!");
            GameController.Instance.ReceiveHealth(gameObject, healthOnPickup);
            //areaOfDetection.radius -= healthOnPickup; disabled for training purposes
            
            if (collectibleList.Contains(collision.gameObject))
            {
                //Debug.Log("I have eaten " + collision.gameObject.name);
                collectibleList.Remove(collision.gameObject);
                //Debug.Log("And there are now " + collectibleList.Count + " in list.");
                collision.gameObject.GetComponent<BoxCollider>().enabled = false;
                collision.gameObject.SetActive(false); 
                //Destroy(collision.gameObject);
            }
            // remove from list then destroy, otherwise null pointer exception
            SetReward(1f);
            Done();
        }

        if (collision.gameObject.CompareTag("Wall"))
        {
            AddReward(-0.1f);
            Done();
        }
```

I am adding a -0.1 reward to the wall in order to avoid the agent sticking too much to the wall and not actually explore the area. Not sure if this should be set instead of add, and also if it should be larger than the hazard in order to give it more incentive to explore.

I now modified this scenario by enlarging the trigger sphere and increasing the curiosity strength.

I am using the following yaml config file with ppo (not sure if I should go for sac in this case). The spikes and collectibles in the map are static. This is a new run.

```
default:
    trainer: ppo
    batch_size: 4096
    beta: 5.0e-3
    buffer_size: 40960
    epsilon: 0.2
    hidden_units: 256
    lambd: 0.95
    learning_rate: 1.0e-4
    learning_rate_schedule: linear
    max_steps: 5.0e6
    memory_size: 512
    normalize: false
    num_epoch: 8
    num_layers: 2
    time_horizon: 1024
    sequence_length: 64
    summary_freq: 1000
    use_recurrent: false
    vis_encode_type: simple
    reward_signals:
        extrinsic:
            strength: 1.0
            gamma: 0.99
        curiosity:
            strength: 0.1
            gamma: 0.90
            encoding_size: 256
```

Any tips on optimizing the rewards and/or the configuration file for ppo in this scenario?

**Comment by firepro20** [comment]
Played around with hyperparameters and got a better result for my case scenario
